templates/VPC/vpc-20-private.yaml

```diff
@@ -0,0 +1,396 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >-
+  Creates a VPC with public and private subnets and a managed NAT gateway.
+Parameters:
+  VpcName:
+    Description: The VPC name (i.e. Synapse-Prod)
+    Type: String
+  VpcSubnetPrefix:
+    Description: The VPC subnet prefix (i.e. 10.255)
+    Type: String
+  PublicSubnetZones:
+    Description: Availability zones for public subnets
+    Type: List<AWS::EC2::AvailabilityZone::Name>
+    ConstraintDescription: List of Availability Zones in a region, such as us-east-1a, us-east-1b, us-east-1c
+    Default: "us-east-1a, us-east-1b, us-east-1c"
+  PrivateSubnetZones:
+    Description: Availability zones for private subnets
+    Type: List<AWS::EC2::AvailabilityZone::Name>
+    ConstraintDescription: List of Availability Zones in a region, such as us-east-1a, us-east-1b, us-east-1c
+    Default: "us-east-1a, us-east-1b, us-east-1c"
+  VpnCidr:
+    Description: CIDR of the (sophos-utm) VPN
+    Type: String
+    Default: "10.1.0.0/16"
+    AllowedPattern: ^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/(1[6-9]|2[0-8]))$
+    ConstraintDescription: CIDR block parameter must be in the form x.x.x.x/16-28
+Mappings:
+  SubnetConfig:
+    VPC:
+      CIDR: "0.0/20"  # 10.255.0.0 - 10.255.15.255
+    Private:
+      CIDR: "0.0/22"  # 10.255.0.0 - 10.255.3.255
+    Private1:
+      CIDR: "4.0/22"  # 10.255.4.0 - 10.255.7.255
+    Private2:
+      CIDR: "8.0/22"  # 10.255.8.0 - 10.255.11.255
+    Private3:
+      CIDR: "12.0/23" # 10.255.12.0 - 10.255.13.255
+    Private4:
+      CIDR: "14.0/24" # 10.255.14.0 - 10.255.14.255
+    Public:
+      CIDR: "15.0/24" # 10.255.15.0 - 10.255.15.255
+Resources:
+  VPC:
+    Type: "AWS::EC2::VPC"
+    Properties:
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      CidrBlock: !Join
+        - '.'
+        - - !Ref VpcSubnetPrefix
+          - !FindInMap [SubnetConfig, VPC, CIDR]
+      Tags:
+        - Key: "Name"
+          Value: !Ref VpcName
+  PublicSubnet:
+    Type: "AWS::EC2::Subnet"
+    Properties:
+      MapPublicIpOnLaunch: true
+      VpcId: !Ref VPC
+      CidrBlock: !Join
```

Comment:
minor, but i would really like to see all these `!Join` converted to `!Sub`

Reply:
Agree but this a bit time consuming and needs to be done throughout a lot of aws-infra templates -- they're showing up here because this stuff was copied and pasted from another VPC template. Don't you have a new intern yet? 😁

Reply:
OMG! We keep asking @brucehoff but he's decided to allocate interns to other projects 😞